cli/ioctl/cmd/wallet/walletcreate.go

```diff
@@ -41,48 +40,29 @@ func init() {
 }
 
 func walletCreate() string {
-	success := "New wallet \"" + name + "\" created, password = " + password +
-		"\n**Remember to save your password. The wallet will be lost if you forgot the password!!"
-	file, err := ioutil.ReadFile(config.DefaultConfigFile)
+	w, err := loadWallets()
 	if err != nil {
-		if os.IsNotExist(err) {
-			// special case of empty config file just being created
-			addr, err := newWallet()
-			if err != nil {
-				return err.Error()
-			}
-			line := walletPrefix + "\n" + name + ":" + addr + "\n" + walletEnd
-			if err := ioutil.WriteFile(config.DefaultConfigFile, []byte(line), 0600); err != nil {
-				return fmt.Sprintf("failed to create config file %s", config.DefaultConfigFile)
-			}
-			return success
-		}
-		return fmt.Sprintf("failed to open config file %s", config.DefaultConfigFile)
+		return err.Error()
 	}
-	// parse the wallet section from config file
-	_, end, find, exist := parseConfig(file, walletPrefix, walletEnd, name)
-	if exist {
-		return "A wallet named " + name + " already exists"
+	if _, ok := w.WalletList[name]; ok {
+		return fmt.Sprintf("A wallet named \"%s\" already exists.", name)
 	}
 	addr, err := newWallet()
 	if err != nil {
 		return err.Error()
 	}
-	// insert a line for the new wallet
-	lines := strings.Split(string(file), "\n")
-	if !find {
-		lines = append(lines, walletPrefix, name+":"+addr, walletEnd)
-	} else {
-		after := make([]string, len(lines)-end)
-		copy(after, lines[end:])
-		lines[end] = name + ":" + addr
-		lines = append(lines[:end+1], after...)
+	w.WalletList[name] = addr
+	out, err := yaml.Marshal(&w)
```

Comment:
the config file also have a line for endpoint, like:
endpoint:localhost:/14014
this line need to be kept and write back to config file
will 'out' here include this line?